Docs: Update labels to fields transform (#30086)
grafana/grafana: docs/sources/panels/transformations/types-options.md

## Labels to fields

> This transform can transform time series results that include labels or tags to a table structure where each label becomes it's own field.

Reviewer: ```suggestion
This transformation changes time series results that include labels or tags into to a table structure where each label becomes its own field.
```

> Given a query result of two time series

Reviewer 2: ```suggestion
Given a query result of two time series:
```

| 2020-07-07 11:34:20 | Server A | EU         | 1     |
| 2020-07-07 11:34:20 | Server B | EU         | 2     |

### Value field name

If you selected Server as the **Value field name**, then you would get one field for every value of the Server label.

| Time                | Datacenter | Server A | Server B |
| ------------------- | ---------- | -------- | -------- |
| 2020-07-07 11:34:20 | EU         | 1        | 2        |

### Merging behavior

> The labels to fields transformer is internally two separate transformations. One that acts on single series (extracting labels to fields) followed by the [merge](#merge) transformation to join all the results into a single table. 

Reviewer 2: ```suggestion
The merge transformation tries to join all matching fields.

The labels to fields transformer is internally two separate transformations. One that acts on single series (extracting labels to fields) followed by the [merge](#merge) transformation to join all the results into a single table.
```

> The labels to fields transformer is internally two separate transformations. One that acts on single series (extracting labels to fields) followed by the [merge](#merge) transformation to join all the results into a single table. 

Reviewer: ```suggestion
The labels to fields transformer consist of two separate transformations. The first acts on single series (extracting labels to fields. The second is the [merge](#merge) transformation to join all the results into a single table. 
```

> To illustrate this, here is another example where you have two queries that return time series with non overallaping labels. 

Reviewer: ```suggestion
To illustrate this, here is an example where you have two queries that return time series with no overlapping labels. 
```

* Series 1: labels Server=ServerA
* Series 2: labels Datacenter=EU

> This will first result in these two tables

Reviewer 2: ```suggestion
This will first result in these two tables:
```

| Time                | Server  | Value | 
| ------------------- | ------- | ----- | 
| 2020-07-07 11:34:20 | ServerA | 10

| Time                | Datacenter | Value | 
| ------------------- | ---------- | ----- |
| 2020-07-07 11:34:20 | EU         | 20

After merge: 

| Time                | Server  | Value | Datacenter |
| ------------------- | ------- | ----- | ---------- |
| 2020-07-07 11:34:20 | ServerA | 10    |            |
| 2020-07-07 11:34:20 |         | 20    | EU         |

> There is currently no option to disable this merge step. The merge transform tries to join 

Reviewer 2: Should this information be before the examples? Something like:

The merge transformation tries to join all matching fields. This merge step is required and cannot be turned off.

Author: Yes, good idea. Moved to before examples
